File: r/native/HelloWorldKomencoNative/Operation.r
# Define the Operation class
Operation <- setRefClass(
  "Operation",
  fields = list(
    gate = "character",
    qubits = "list",
    params = "list"
  ),
  methods = list(
    initialize = function(gate, qubits, params) {
      .self$gate <- gate
      .self$qubits <- qubits
      .self$params <- params
    },
    getGate = function() {
      return(.self$gate)
    },
    getQubits = function() {
      return(.self$qubits)
    },
    getParams = function() {
      return(.self$params)
    }
  )
)

File: f#/native/HelloWorldKomencoNative/Operation.fs
namespace Automatski.Komenco

open System.Collections.Generic

type Operation(gate: string, qubits: List<int>, parameters: List<double>) =
    member this.Gate = gate
    member this.Qubits = qubits
    member this.Params = parameters

    member this.GetGate() = this.Gate
    member this.GetQubits() = this.Qubits
    member this.GetParams() = this.Params